### PR TITLE
daemon: SystemInfo() extract collecting data to more helper functions

### PR DESCRIPTION
**Pull request by thaJeztah**
This makes it more inline with other data we collect, and can be used to make some info optional at some point to make the `/info` endpoint more modular (instead of one heavy operation to collect all data possible).

### daemon: unexport Daemon.ID and Daemon.RegistryService

These are used internally only, and set by daemon.NewDaemon(). If they're
used externally, we should add an accessor added (which may be something
we want to do for daemon.registryService (which should be its own backend)

### daemon: SystemInfo() extract collecting debugging information to a helper

This makes it more inline with other data we collect, and can be used to make
some info optional at some point.

fillDebugInfo sets the current debugging state of the daemon, and additional
debugging information, such as the number of Go-routines, and file descriptors.

Note that this currently always collects the information, but the CLI only
prints it if the daemon has debug enabled. We should consider to either make
this information optional (cli to request "with debugging information"), or
only collect it if the daemon has debug enabled. For the CLI code, see
https://github.com/docker/cli/blob/v20.10.12/cli/command/system/info.go#L239-L244

```go
if info.Debug {
	fmt.Fprintln(dockerCli.Out(), "  File Descriptors:", info.NFd)
	fmt.Fprintln(dockerCli.Out(), "  Goroutines:", info.NGoroutines)
	fmt.Fprintln(dockerCli.Out(), "  System Time:", info.SystemTime)
	fmt.Fprintln(dockerCli.Out(), "  EventsListeners:", info.NEventsListener)
}
```

Additional note: the CLI considers info.SystemTime debugging information. This
felt a bit "odd" (daemon time could be useful for standard use), so I left this
out of this function.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like I had this one stashed from when I was looking at https://github.com/moby/moby/pull/43245

**Comment by thaJeztah** [comment]
oh, actually, I had more in my stash; let me have a look

**Comment by thaJeztah** [comment]
CI failure is unrelated. Still not sure why we sometimes run into this (test should detect if it's supported?)

```
=== RUN   TestDevmapperIncreaseLoopBackSize
time="2022-03-01T22:58:29Z" level=error msg="Udev sync is not supported. This will lead to data loss and unexpected behavior. Install a more recent version of libdevmapper or select a different storage driver. For more information, see https://docs.docker.com/engine/reference/commandline/dockerd/#storage-driver-options" storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=warning msg="Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man dockerd` to refer to dm.thinpooldev section." storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=info msg="Creating filesystem xfs on device docker-0:60-6187-base, mkfs args: [/dev/mapper/docker-0:60-6187-base]" storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=info msg="Successfully created filesystem xfs on device docker-0:60-6187-base" storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=warning msg="unmountAndDeactivate: open /tmp/docker-graphtest-1124247676/devicemapper/mnt: no such file or directory" storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=error msg="Udev sync is not supported. This will lead to data loss and unexpected behavior. Install a more recent version of libdevmapper or select a different storage driver. For more information, see https://docs.docker.com/engine/reference/commandline/dockerd/#storage-driver-options" storage-driver=devicemapper
time="2022-03-01T22:58:29Z" level=warning msg="Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev` or use `man dockerd` to refer to dm.thinpooldev section." storage-driver=devicemapper
time="2022-03-01T22:58:30Z" level=warning msg="Base device already exists and has filesystem xfs on it. User specified filesystem  will be ignored." storage-driver=devicemapper
    devmapper_test.go:124: data or metadata loop back size is incorrect
time="2022-03-01T22:58:30Z" level=warning msg="unmountAndDeactivate: open /tmp/docker-graphtest-1124247676/devicemapper/mnt: no such file or directory" storage-driver=devicemapper
--- FAIL: TestDevmapperIncreaseLoopBackSize (0.35s)
```

File: api/types/types.go [hunk 1]
```diff
@@ -220,6 +217,15 @@ type Info struct {
 	Warnings []string
 }
 
+// ContainerStateInfo holds the total number of containers, and the number of
+// containers for each state (running, paused, and stopped).
+type ContainerStateInfo struct {
```
Comment (tianon) [inline api/types/types.go:222]: I'm not sure I understand what this specific part of this change does for us? :see_no_evil:

(As far as I can tell, it just makes everywhere we use it more verbose :sweat_smile:)

Comment (thaJeztah) [reply api/types/types.go:222]: Ah, yes, that's fair. I think I was playing with some variations here (considering to leave things `nil` if we don't set them.

But let me drop that commit, and keep that for when we're actually gonna make things more granular

Comment (thaJeztah) [reply api/types/types.go]: OK; removed the commit 👍

**Review by tianon — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**